.azure-pipelines/steps/run-tests-windows.yml

```diff
@@ -8,10 +8,62 @@ steps:
     versionSpec: '$(python.version)'
     architecture: '$(python.architecture)'
 
+- powershell: |
+    echo "Installing FS-iSCSITarget-Server"
+    Install-WindowsFeature -Name FS-iSCSITarget-Server
+
+    # Allow connection on loopback interface.
+    Set-ItemProperty -Path 'HKLM:\SOFTWARE\Microsoft\iSCSI Target' -Name AllowLoopBack -Value 1
+
+    echo "Starting MsiSCSI"
+    Start-Service msiscsi
+    $retry = 10
+    do
+    {
+      $count = (Get-Service msiscsi | ? {$_.status -eq "Running"}).count
+      $retry--
+      sleep -Milliseconds 500
+    } until ($count -Eq 0 -or $retry -Eq 0)
+
+    $service = Get-Service msiscsi
+    if ($service.Status -Ne "Running") {
+      throw "msiscsi not running"
+    }
+
+    echo "Configuring Firewall"
+    Get-NetFirewallServiceFilter -Service msiscsi | Enable-NetFirewallRule
+
+    echo "Configuring RAMDisk"
+    New-IscsiVirtualDisk -ComputerName localhost -Path ramdisk:RAMDISK1.vhdx -Size 1GB
+    New-IscsiServerTarget Target1 -ComputerName localhost -InitiatorId "IQN:*"
+    Add-IscsiVirtualDiskTargetMapping -ComputerName localhost -TargetName Target1 -Path ramdisk:RAMDISK1.vhdx -Lun 1
+
+    echo "Connecting to iSCSI"
+    $ip = (Get-NetIPAddress -AddressFamily IPv4).ipaddress[0]
```

Comment:
Not sure how reliable the ordering of addresses is - this has to resolve to the non-loopback address which in my testing always ended up first.